cmd/terramate/cli/script_tree.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/hcl"
+	project2 "github.com/terramate-io/terramate/project"
+	"golang.org/x/exp/slices"
+)
+
+func (c *cli) printScriptTree() {
+	srcpath := project2.PrjAbsPath(c.rootdir(), c.wd())
+
+	cfg, found := c.cfg().Lookup(srcpath)
+	if !found {
+		return
+	}
+
+	rootNode, topNode := addParentScriptTreeNodes(cfg, nil, true)
+	addChildScriptTreeNodes(cfg, topNode)
+
+	var sb strings.Builder
+	rootNode.format(&sb, "", nil)
+	c.output.MsgStdOut(sb.String())
+}
+
+type scriptsTreeNode struct {
+	DirName  string
+	IsStack  bool
+	Scripts  []*hcl.Script
+	Children []*scriptsTreeNode
+	Parent   *scriptsTreeNode
+	Visible  bool
+}
+
+func (node *scriptsTreeNode) format(w io.Writer, prefix string, parentScripts []string) {
+	stackColor := color.New(color.FgGreen).SprintFunc()
+	scriptColor := color.New(color.FgYellow).SprintFunc()
+	parentScriptColor := color.New(color.Faint).SprintFunc()
+
+	var text string
+	if node.DirName != "" {
+		text += node.DirName
+	} else {
+		text += "/"
+	}
+
+	if node.IsStack {
+		fmt.Fprintln(w, "#"+stackColor(text))
+	} else {
+		fmt.Fprintln(w, text)
+	}
+
+	visibleChildren := make([]*scriptsTreeNode, 0, len(node.Children))
+	for _, child := range node.Children {
+		if child.Visible {
+			visibleChildren = append(visibleChildren, child)
+		}
+	}
+
+	blockPrefix := prefix
+	if len(visibleChildren) > 0 {
+		blockPrefix += "│ "
+	} else {
+		blockPrefix += "  "
+	}
+
+	for _, sc := range node.Scripts {
+		desc := formatScriptDescription(sc)
+		fmt.Fprintln(w, blockPrefix+"* "+scriptColor(sc.Name()+": "+desc))
+	}
+
+	if node.IsStack {
+		for _, p := range parentScripts {
+			_, found := slices.BinarySearchFunc(node.Scripts, p,
```

Comment:
I don't expect more than 10 scripts per directory, so I think `slices.ContainsFunc()` could even perform better. I like to follow Rob Pike style rules for complexity defined [here](https://www.lysator.liu.se/c/pikestyle.html), check `rule 3`.